src/pools/D3MCompoundPool.sol

```diff
@@ -0,0 +1,199 @@
+        emit Deny(usr);
+    }
+
+    function file(bytes32 what, address data) external auth {
+        if (what == "king") king = data;
+        else revert("D3MCompoundPool/file-unrecognized-param");
+        emit File(what, data);
+    }
+
+    function hope(address hub) external override auth {
+        CanLike(D3mHubLike(hub).vat()).hope(hub);
+    }
+
+    function nope(address hub) external override auth {
+        CanLike(D3mHubLike(hub).vat()).nope(hub);
+    }
+
+    function deposit(uint256 wad) external override auth returns (bool) {
+        uint256 prev = cDai.balanceOf(address(this));
+        require(cDai.mint(wad) == 0, "D3MCompoundPool/mint-failure");
+        // As interest was accrued on `mint` we can use the non accruing `exchangeRateStored`
+        require(
+            cDai.balanceOf(address(this)) ==
+            prev + _wdiv(wad, cDai.exchangeRateStored()), "D3MCompoundPool/incorrect-cdai-credit"
+        );
+        return true;
+    }
+
+    function withdraw(uint256 wad) external override auth returns (bool) {
+        require(cDai.redeemUnderlying(wad) == 0, "D3MCompoundPool/redeemUnderlying-failure");
+        dai.transfer(msg.sender, wad);
+        return true;
+    }
+
+    function transfer(address dst, uint256 wad) external override auth returns (bool) {
+        return cDai.transfer(dst, _wdiv(wad, cDai.exchangeRateCurrent()));
+    }
+
+    function transferAll(address dst) external override auth returns (bool) {
+        return cDai.transfer(dst, cDai.balanceOf(address(this)));
+    }
+
+    function preDebtChange(bytes32) external override {
+        require(cDai.accrueInterest() == 0, "D3MCompoundPool/accrueInterest-failure");
+    }
+
+    function postDebtChange(bytes32) external override {}
+
+    // Does not accrue interest (as opposed to cToken's balanceOfUnderlying() which is not a view function).
+    function assetBalance() public view override returns (uint256) {
+        (uint256 error, uint256 cTokenBalance,, uint256 exchangeRate) = cDai.getAccountSnapshot(address(this));
+        require(error == 0, "D3MCompoundPool/getAccountSnapshot-failure");
+        return _wmul(cTokenBalance, exchangeRate);
+    }
+
+    function maxDeposit() external pure override returns (uint256) {
+        return type(uint256).max;
+    }
+
+    function maxWithdraw() external view override returns (uint256) {
+        return _min(cDai.getCash(), assetBalance());
+    }
+
+    // TODO: remove active from pool and rely just on active in plan? - once supported in the interface
+    function active() external pure override returns (bool) {
+        return true;
+    }
+
+    function collect() external {
+        require(king != address(0), "D3MCompoundPool/king-not-set");
+
+        address[] memory holders = new address[](1);
+        holders[0] = address(this);
+        address[] memory cTokens = new address[](1);
+        cTokens[0] = address(cDai);
+
+        comptroller.claimComp(holders, cTokens, false, true);
```

Comment:
Consider fetching the comptroller from cDai everytime here instead of having it in storage.

Reply:
I eventually opted out of this and even make `comp` immutable.
Since we later have `comp.transfer(king, amt)` I feel it's more secured to have both `comptroller` and `comp` immutable (otherwise a malicious comptroller can allow sending out cdai, even though that line should send to the pause proxy).

@iamchrissmith fyi (I think in aave the situation with `rewardsClaimer` is less risky as there is no additional transfer step).